### PR TITLE
Device tracking JavaScript

### PR DESCRIPTION
**Pull request by evcodes**
*Issue #, if available:*
In the Amplify-js repo we have issue [#1335](https://github.com/aws-amplify/amplify-js/issues/1355) and [#7688](https://github.com/aws-amplify/amplify-js/issues/7688)

*Description of changes:*
Device tracking documentation in the authentication library for JavaScript. This documentation is suppose to match the documentation that exists within iOS and Android however, the implementation details are different.

I will be merging this in once we get an approval from internal stakeholders

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by sammartinez** [comment]
Thanks @medelman17, this feature is still being worked on. At the moment, we are dependant of the feature being complete prior to merging this in.

**Comment by sammartinez** [comment]
@evcodes Can you flip this to a draft until we are ready to go live?

**Comment by medelman17** [comment]
Thanks, @sammartinez!

**Review by renebrandel — CHANGES_REQUESTED** [review]
The content is an exact copy of https://github.com/aws-amplify/docs/blob/main/docs/lib/auth/fragments/native_common/device_features/common.md

Please the content into `common/*` and only insert JS related code snippets when needed. Avoid duplication.

**Comment by evcodes** [comment]
> The content is an exact copy of https://github.com/aws-amplify/docs/blob/main/docs/lib/auth/fragments/native_common/device_features/common.md
> 
> Please the content into `common/*` and only insert JS related code snippets when needed. Avoid duplication.

just curious, how come ios and android have duplicated code when it's _exactly_ the same as well?

**Review by renebrandel — APPROVED** [review]

File: docs/lib/auth/fragments/js/device_features/10_rememberDevice.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+```javascript
+async function rememberDevice() {
+    try{
+        const result = await Amplify.Auth.rememberDevice()
```
Comment (elorzafe) [inline docs/lib/auth/fragments/js/device_features/10_rememberDevice.md:4]:
```suggestion
        const result = await Auth.rememberDevice();
```

File: docs/lib/auth/fragments/js/device_features/20_forgetDevice.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+```javascript
+async function forgetDevice() {
+    try{
+        const result = await Amplify.Auth.forgetDevice()
```
Comment (elorzafe) [inline docs/lib/auth/fragments/js/device_features/20_forgetDevice.md:4]:
```suggestion
        const result = await Auth.forgetDevice();
```

File: docs/lib/auth/fragments/js/device_features/30_fetchDevice.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+```javascript
+async function fetchDevices() {
+    try{ 
+        const result = await Amplify.Auth.fetchDevices()
```
Comment (elorzafe) [inline docs/lib/auth/fragments/js/device_features/30_fetchDevice.md:4]:
```suggestion
        const result = await Auth.fetchDevices()
```

**Review by elorzafe — COMMENTED** [review]
I would change `Amplify.Auth.` to just `Auth.`

**Review by sammartinez — COMMENTED** [review]
Please update the callouts from what Francisco stated and I am good.

File: docs/lib/auth/fragments/js/emailpassword.md [hunk 4]
```diff
@@ -23,7 +23,7 @@ async function signUp() {
 }
 ```
 
-The `Auth.signUp` promise returns a data object of type [`ISignUpResult`](https://github.com/aws-amplify/amplify-js/blob/4644b4322ee260165dd756ca9faeb235445000e3/packages/amazon-cognito-identity-js/index.d.ts#L136-L139) with a [`CognitoUser`](https://github.com/aws-amplify/amplify-js/blob/4644b4322ee260165dd756ca9faeb235445000e3/packages/amazon-cognito-identity-js/index.d.ts#L48). 
+The `Auth.signUp` promise returns a data object of type [`ISignUpResult`](https://github.com/aws-amplify/amplify-js/blob/4644b4322ee260165dd756ca9faeb235445000e3/packages/amazon-cognito-identity-js/index.d.ts#L136-L139) with a [`CognitoUser`](https://github.com/aws-amplify/amplify-js/blob/4644b4322ee260165dd756ca9faeb235445000e3/packages/amazon-cognito-identity-js/index.d.ts#L48) where the userSub is a universally unique identifier (UUID) of the authenticated user. This is not the same as username.
```
Comment (elorzafe) [inline docs/lib/auth/fragments/js/emailpassword.md:26]:
```suggestion
The `Auth.signUp` promise returns a data object of type [`ISignUpResult`](https://github.com/aws-amplify/amplify-js/blob/4644b4322ee260165dd756ca9faeb235445000e3/packages/amazon-cognito-identity-js/index.d.ts#L136-L139) with a [`CognitoUser`](https://github.com/aws-amplify/amplify-js/blob/4644b4322ee260165dd756ca9faeb235445000e3/packages/amazon-cognito-identity-js/index.d.ts#L48). `CognitoUser` contains a `userSub` which is a unique identifier of the authenticated user; the `userSub` is not the same as the `username`.
```

Comment (sammartinez) [reply docs/lib/auth/fragments/js/emailpassword.md:26]: @evcodes Lets look to commit this change

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

File: docs/lib/auth/fragments/js/emailpassword.md (hunk 4 above)
Comment (evcodes) [reply docs/lib/auth/fragments/js/emailpassword.md]: all ready

**Review by elorzafe — APPROVED** [review]
Thanks @evcodes 🌮 🎉 🏅

**Pull request merged**